### PR TITLE
Add Xunit adapter to enable testing in VS

### PR DESCRIPTION
**Pull request by genlu**

### CODE REVIEW

File: build/Rulesets/NonShippingProject_BuildRules.ruleset [hunk 1]
```diff
@@ -33,4 +33,17 @@
     <!-- Mark constant field as 'const' instead of static readonly - not useful for tests -->
     <Rule Id="CA1802" Action="None" />
   </Rules>  
+
+  <Rules AnalyzerId="Xunit.Analyzers" RuleNamespace="Xunit.Analyzers">
+    <!-- Do not use Enumerable.Any() to check if a value exists in a collection -->
+    <Rule Id="xUnit2012" Action="None" />
```
Comment (jmarolf) [inline build/Rulesets/NonShippingProject_BuildRules.ruleset]: Can we file bugs for all of these cases?  Seems like these are hiding real issues

Comment (genlu) [reply build/Rulesets/NonShippingProject_BuildRules.ruleset]: Sure.

**Comment by sharwell** [comment]
:memo: I'm halfway through the day. Subjective evaluation is showing a consistent move towards not being viable. While it's possible that something could change and the painful parts of the experience get avoided, at this point it appears unlikely.

**Comment by ManishJayaswal** [comment]
@sharwell - are you trying this on preview 3 builds?

**Comment by sharwell** [comment]
I'm testing on 15.6 Preview 2. I'll follow up after my work is complete with the observed (objective) results.

**Comment by sharwell** [comment]
This is still really bad on Preview 3. I'm still in the process of investigating the cause so things can be improved, but we're going to have to move this PR to Preview 4 at the earliest.

**Comment by sharwell** [comment]
@jasonmalinowski @Pilchie Can you review the updated documentation in the final commit of this PR? I'm planning to merge on successful build and notify the team. 😄

File: build/Targets/Imports.targets [hunk 2]
```diff
@@ -356,21 +356,21 @@
...
+    Both of these include xunit.abstractions.dll in different forms: None and reference
+    respectively.  This creates a double write violation in our build.  The None item
     is unnecessary for us since we reference the xunit package hence remove it here. 
 
     https://github.com/dotnet/roslyn/issues/18753
   -->
   <Target Name="RemoveDuplicateXUnitContent">
       <ItemGroup>
-        <Content Remove="@(Content)" Condition="'%(Filename)%(Extension)' == 'xunit.abstractions.dll'" />
+        <None Remove="@(None)" Condition="'%(Filename)%(Extension)' == 'xunit.abstractions.dll'" />
```
Comment (jasonmalinowski) [inline build/Targets/Imports.targets]: @sharwell Is this still needed? Content would have caused a double write, but None should have been ignored?

Comment (sharwell) [reply build/Targets/Imports.targets]: I believe it is still necessary.

https://github.com/xunit/xunit/blob/f32d5fd70e821fc9ccc807878cba2577d68f6921/src/xunit.runner.visualstudio/build/xunit.runner.visualstudio.desktop.props#L16

**Pull request merged**